mathics/builtin/__init__.py

```diff
@@ -103,13 +103,39 @@ def get_module_doc(module):
     return title, text
 
 
+inout_hooks = {'$PreRead':"$PreRead is a global variable whose value, if set, is applied to the \
+text or box form of every input expression before it is fed to the parser. (Not implemented yet) ",
+               '$Pre': "$Pre is a global variable whose value, if set, is applied to every \
+input expression.",
+               '$Post': "$Post is a global variable whose value, if set, is applied to every \
+output expression. ",
+               '$PrePrint': "$PrePrint is a global variable whose value, if set, is applied to \
+every expression before it is printed. ",
+               '$SyntaxHandler': "$SyntaxHandler is a global variable which, if set, is applied to any \
+               input string that is found to contain a syntax error. (Not implemented yet) ",
+           }
+
+def load_InOut_hooks(definitions):
+    from mathics.core.definitions import Definition
+    from mathics.core.expression import  Symbol, String
+    from mathics.core.rules import Rule
+    from mathics.core.pattern import Pattern_create
+    from mathics.builtin.inout import MessageName
+    for hook in inout_hooks:
+        fullname = "System`" + hook
+        definitions.builtin[fullname] = Definition(name = fullname )
+        usage = Rule(MessageName(Symbol(fullname), String("usage")), String(inout_hooks[hook]))
+        definitions.builtin[fullname].set_values_list('messages',[usage])
```

Comment:
Why do the hooks need a special loading mechanism? Does the standard `Builtin` method not work?